userspace/libsinsp/unix_paths.cpp

```diff
@@ -142,9 +144,13 @@ void copy_and_sanitize_path(char* target, char* targetbase, const char* path, ch
 			else if(*pc == separator)
 			{
 				//
-				// separator, if the last char is already a separator, skip it
+				// separator:
+				// * if the last char is already a separator, skip it
+				// * if we are back at targetbase but targetbase was not empty before, it means we
+				//   fully rewinded back to targetbase and the string is now empty. Skip separator.
+				//   Example: "/foo/../a" -> "/a" BUT "foo/../a" -> "a"
 				//
-				if(tc > targetbase && *(tc - 1) == separator)
+				if((tc > targetbase && *(tc - 1) == separator) || (tc == targetbase && !empty_base))
```

Comment:
When cutting away `../` in the target string, we return to its start, avoid putting a `/`; this fixes edge cases like:
```
"foo/../a"
```
that previously returned `/a` instead of just `a`.

Reply:
This behavior is covered by tests.